src/UnitfulRecipes.jl

```diff
@@ -163,9 +163,35 @@ function append_unit_if_needed!(attr, key, label::Nothing, u)
     attr[key] = UnitfulString(string(u), u)
 end
 function append_unit_if_needed!(attr, key, label::S, u) where {S <: AbstractString}
-    if label ≠ ""
-        attr[key] = UnitfulString(S(string(label, " (", u, ")")), u)
+    if !isempty(label)
+        attr[key] = UnitfulString(S(surround(label,u, get(attr,:surround,:round))), u)
     end
 end
 
+#=============================================
+Surround unit string with specified delimiters
+=============================================#
+surround(l,u,f::Nothing) = string(l,' ',u)
+surround(l,u,f::Function) = f(l,u)
+surround(l,u,f::AbstractString) = string(l,f,u)
+surround(l,u,f::NTuple{2,<:AbstractString}) = string(l,f[1],u,f[2])
+surround(l,u,f::NTuple{3,<:AbstractString}) = string(f[1],l,f[2],u,f[3])
+surround(l,u,f::Char) = string(l,' ',f,' ',u)
+surround(l,u,f::NTuple{2,Char}) = string(l,' ',f[1],u,f[2])
+surround(l,u,f::NTuple{3,Char}) = string(f[1],l,' ',f[2],u,f[3])
+surround(l,u,f::Bool) = f ? surround(l,u,:round) : surround(l,u,nothing)
```

Comment:
This looks neat and exhaustive: nice work! Otherwise, spaces after commas :) :
```julia
surround(l, u, f::Nothing) = string(l, ' ', u)
surround(l, u, f::Function) = f(l, u)
surround(l, u, f::AbstractString) = string(l, f, u)
surround(l, u, f::NTuple{2,<:AbstractString}) = string(l, f[1], u, f[2])
surround(l, u, f::NTuple{3,<:AbstractString}) = string(f[1], l, f[2], u, f[3])
surround(l, u, f::Char) = string(l, ' ', f, ' ', u)
surround(l, u, f::NTuple{2,Char}) = string(l, ' ', f[1], u, f[2])
surround(l, u, f::NTuple{3,Char}) = string(f[1], l, ' ', f[2], u, f[3])
surround(l, u, f::Bool) = f ? surround(l, u, :round) : surround(l, u, nothing)
```